sql/parser/sql.y

```diff
@@ -1402,11 +1402,11 @@ rename_stmt:
   }
 | ALTER INDEX qualified_name RENAME TO name
   {
-    $$ = nil
+    $$ = &RenameIndex{Name: $3, NewName: Name($6), IfExists: false}
   }
```

Comment:
[postgres](http://www.postgresql.org/docs/9.4/static/sql-altertable.html):
`ALTER TABLE [ IF EXISTS ] [ ONLY ] name [ * ]
    RENAME CONSTRAINT constraint_name TO new_constraint_name`

[mysql](https://dev.mysql.com/doc/refman/5.7/en/alter-table.html):
`ALTER [IGNORE] TABLE tbl_name RENAME {INDEX|KEY} old_index_name TO new_index_name`

I'm find with having `ALTER INDEX` only for now, but add a TODO to implement the `ALTER TABLE` part of it.